plugins/modules/nd_backup_restore.py

```diff
@@ -112,49 +145,99 @@ def main():
-        elif len(restored_info) == 1:
-            restore_key = restored_info[0].get("key")
-        elif restore_key is not None and restore_key not in [file_dict.get("key") for file_dict in restored_info]:
-            nd.fail_json(
-                "Provided key for the restore '{0}' not found."
-                " Please provide a valid restore key by querying all the restored jobs and looking up the desired restore key.".format(name)
-            )
-        nd.existing = next((file_dict for file_dict in restored_info if file_dict.get("key") == restore_key), {})
-    else:
-        nd.existing = restored_objs
-
-    nd.previous = nd.existing
+    ignore_persistent_ips = nd.params.get("ignore_persistent_ips")
+    restore_type = BACKUP_TYPE.get(nd.params.get("restore_type"))
+    remote_location = nd.params.get("remote_location")
+
+    backup_status_path = "/api/config/class/imports" if nd.version < "3.2.1" else "/api/action/class/backuprestore/status"
+    backup_status = nd.query_obj(backup_status_path)
+
+    if backup_status.get("operation") == "restore":
+        nd.existing = backup_status
+
+    import_path = "/nexus/infra/api/platform/v1/imports" if nd.version < "3.2.1" else "/api/action/class/backuprestore/restore/file-import"
+
+    if nd.version < "3.2.1":
+        if name:
+            restored_info = [file_dict for file_dict in nd.existing if file_dict.get("description") == name]
+            if len(restored_info) > 1 and restore_key is None and encryption_key is None:
+                nd.fail_json("Multiple restore jobs with the name '{0}' found. Please provide a restore key for the corresponding restored job.".format(name))
+            elif len(restored_info) == 1:
+                restore_key = restored_info[0].get("key")
+            elif restore_key is not None and restore_key not in [file_dict.get("key") for file_dict in restored_info]:
+                nd.fail_json(
+                    "Provided key for the restore '{0}' not found."
+                    " Please provide a valid restore key by querying all the restored jobs and looking up the desired restore key.".format(name)
+                )
+            nd.existing = next((file_dict for file_dict in restored_info if file_dict.get("key") == restore_key), {})
+
+    nd.previous = copy.deepcopy(nd.existing)
 
     if state == "absent":
-        if nd.existing:
-            if not module.check_mode:
-                nd.request("{0}/{1}".format(path, restore_key), method="DELETE")
-            nd.existing = {}
-    elif state == "restore":
-        nd.previous = nd.existing = {}
-
-        payload = {
-            "description": name,
-            "password": encryption_key,
-        }
+        if not module.check_mode:
+            nd.request("{0}/{1}".format(import_path, restore_key) if nd.version < "3.2.1" else import_path, method="DELETE")
```

Comment:
You don't check if restore_key is not None. What is the behavior when none? (if name is not provided for version <3.2.1